cluster-autoscaler/cloudprovider/hetzner/hetzner_node_group.go

```diff
@@ -392,10 +412,16 @@ func createServer(n *hetznerNodeGroup) error {
 		return err
 	}
 
+	cloudInit := n.manager.clusterConfig.LegacyConfig.CloudInit
+
+	if n.manager.clusterConfig.IsUsingNewFormat {
+		cloudInit = n.manager.clusterConfig.NodeConfigs[n.id].CloudInit
+	}
+
 	StartAfterCreate := true
 	opts := hcloud.ServerCreateOpts{
 		Name:             newNodeName(n),
-		UserData:         n.manager.cloudInit,
+		UserData:         string(cloudInit),
```

Comment:
`cloudInit` is already a string

```suggestion
		UserData:         cloudInit,
```